docs/guides/connecting_to_your_data/filesystem/spark.md

```diff
@@ -2,4 +2,152 @@
 title: How to connect to data on a filesystem using Spark
 ---
 
-**TODO** This is a stub
+import NextSteps from '../components/next_steps.md'
+import Congratulations from '../components/congratulations.md'
+import Prerequisites from '../components/prerequisites.jsx'
+import WhereToRunCode from '../components/where_to_run_code.md'
+import SparkDataContextNote from '../components/spark_data_context_note.md'
+import SparkAdditionalNotes from '../components/spark_additional_notes.md'
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+This guide will help you connect to your data stored on a filesystem using spark.
+This will allow you to validate and explore your data.
+
+<Prerequisites>
+
+- Have a access to a working spark installation
+- Have access to data on a filesystem
+
+</Prerequisites>
+
+## Steps
+
+### 1. Choose how to run the code in this guide
+
+<WhereToRunCode />
+
+### 2. 💡 Instantiate your project's DataContext
+
+Import these necessary packages and modules.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L1-L4
+```
+
+<SparkDataContextNote />
+
+Please proceed only after you have instantiated your `DataContext`.
+
+### 3. Configure your Datasource
+
+Using this example configuration, add in your path to a directory that contains some of your data:
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L20-L38
+```
+
+Run this code to test your configuration.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L44
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_python_example.py#L21-L39
+```
+
+Run this code to test your configuration.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_python_example.py#L45
+```
```

Comment:
I think this line needs to be double-checked

Reply:
Good catches!